bodhi/client/__init__.py

```diff
@@ -84,6 +84,39 @@ def new(user, password, **kwargs):
 
 
 @updates.command()
+@click.argument('builds')
+@click.option('--user', envvar='USERNAME')
+@click.option('--password', prompt=True, hide_input=True)
+@click.option('--type', default='bugfix', help='Update type', required=True,
+              type=click.Choice(['security', 'bugfix',
+                                 'enhancement', 'newpackage']))
+@click.option('--notes', help='Update description')
+@click.option('--notes-file', help='Update description from a file')
+@click.option('--bugs', help='Comma-seperated list of bug numbers', default='')
+@click.option('--close-bugs', default=True, is_flag=True, help='Automatically close bugs')
+@click.option('--request', help='Requested repository',
+              type=click.Choice(['testing', 'stable', 'unpush']))
+@click.option('--autokarma', default=True, is_flag=True, help='Enable karma automatism')
+@click.option('--stable-karma', type=click.INT, help='Stable karma threshold')
+@click.option('--unstable-karma', type=click.INT, help='Unstable karma threshold')
+@click.option('--suggest', help='Post-update user suggestion',
+              type=click.Choice(['logout', 'reboot']))
+@click.option('--file', help='A text file containing all the update details')
+@click.option('--staging', help='Use the staging bodhi instance',
+              is_flag=True, default=False)
```

Comment:
It might be possible to create a new Python function decorator that adds all of these flags. Then you can just use the new decorator on this function and the new function, for at least the flags they have in common. That might be the easiest way to avoid duplicating code.